R/data-helpers.R

```diff
@@ -240,7 +240,10 @@ order_data <- function(data, bterms) {
       tv <- seq_rows(data)
     }
     if (any(duplicated(data.frame(gv, tv)))) {
-      stop2("Time points within groups must be unique.")
+      if (!parameterize_ac_effects(bterms)) {
```

Comment:
why do we need to distinquish `has_ac_latent_residuals` and `parameterize_ac_effects`? Naively I would think that the former should be enough but I may be mistaken of course.

Reply:
We probably don't. I think my intent here was to avoid "overwriting" existing latent variable functionality with the new code.

Reply:
I would like to have the latent feature tightly integrated with the rest of the autocorrelation features and code base. Otherwise, we have multiple things that do basically the same and the branching logic becomes confusing. I am sorry that I am so particular about these things. You can let me know when there are changes that I suggest that you would prefer me to do in this PR. I know it is a bit tedious sometimes.